crates/cargo-wdk/src/actions/build/mod.rs

```diff
@@ -375,35 +364,31 @@ impl<'a> BuildAction<'a> {
             return Ok(());
         }
 
-        debug!("Creating the driver package in the target directory");
-        let driver_model = wdk_metadata.driver_model.clone();
-        let target_arch = match self.target_arch {
-            TargetArch::Default(arch) | TargetArch::Selected(arch) => arch,
+        // Resolve the target architecture for the packaging task
+        let target_arch = if let Some(arch) = self.target_arch {
+            arch
+        } else {
+            self.probe_target_arch_from_cargo_rustc(working_dir)?
         };
         debug!("Target architecture for package: {package_name} is: {target_arch}");
-        let mut target_dir = target_dir.to_path_buf();
-        if let TargetArch::Selected(arch) = self.target_arch {
-            target_dir = target_dir.join(to_target_triple(arch));
-        }
-        target_dir = match self.profile {
-            Some(Profile::Release) => target_dir.join("release"),
-            _ => target_dir.join("debug"),
-        };
-        debug!(
-            "Target directory for package: {} is: {}",
-            package_name,
-            target_dir.display()
-        );
+
+        // Set up the `PATH` system environment variable with WDK/SDK bin and tools
+        // paths.
+        wdk_build::cargo_make::setup_path().map_err(|e| {
+            debug!("Failed to set up PATH for WDK/SDK tools");
+            BuildActionError::WdkBuildConfig(e)
+        })?;
+        debug!("PATH env variable is set with WDK bin and tools paths");
```

Comment:
Why do we need to set this environment variable no when we didn't need it earlier? If it's needed for tests, better set it in the test code.